### PR TITLE
Thao feature post form

### PR DESCRIPTION
**Pull request by thaotpnguyen**
In this PR:
- Temporarily parked the component on main page to test

Working feature / buttons:
- Create a placeholder for necessary field
- Next / Back buttons
- Switch for Name Your Own Price 
- Calendar for input date

Button/ Feature not working:

- No "Home" button to go back from step 1 to 0.
- "Finish" button doesn't work.
- Haven't implemented restriction or validation - even the field is required, you can still click next when it's not filled.

### CODE REVIEW

File: client/src/postItem/postItem.jsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      step: step + 1
+    });
+  }
+
+  handleChange (input) {
+    return (e)=> {
+      if (input === 'nameYourOwnPrice') {
+        this.setState ({ [input]: e.target.checked });
+      } else {
+        this.setState ({ [input]: e.target.value }, () => { console.log(this.state); });
```
Comment (joemitz) [inline client/src/postItem/postItem.jsx]: What is this console.log for? Is it for errors?

File: client/src/postItem/postItem.jsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    const { step } = this.state;
+    this.setState ({
+      step: step + 1
+    });
+  }
+
+  handleChange (input) {
+    return (e)=> {
+      if (input === 'nameYourOwnPrice') {
+        this.setState ({ [input]: e.target.checked });
```
Comment (joemitz) [inline client/src/postItem/postItem.jsx]: Is this using destructuring for the key? I've seen a key in square brackets before but I don't know what it means

**Review by joemitz — APPROVED** [review]
Great work Thao. Your code is very clean and easy to read. I put a couple questions but I approve the PR.

File: client/src/postItem/postItem.jsx (hunk 1 above)
Comment (thaotpnguyen) [reply client/src/postItem/postItem.jsx]: I removed it. I was just checking to see if handle change update correctly.

File: client/src/postItem/postItem.jsx (hunk 2 above)
Comment (thaotpnguyen) [reply client/src/postItem/postItem.jsx]: It was the short for initializing an empty object. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names

Comment (joemitz) [reply client/src/postItem/postItem.jsx]: Oh that's very useful, thanks

**Comment by joemitz** [comment]
Looks good

**Pull request merged**